main.go

```diff
@@ -92,14 +100,41 @@ type DNSRRValidator struct {
 	FailIfNotMatchesRegexp []string `yaml:"fail_if_not_matches_regexp"`
 }
 
-var Probers = map[string]func(string, http.ResponseWriter, Module) bool{
-	"http": probeHTTP,
-	"tcp":  probeTCP,
-	"icmp": probeICMP,
-	"dns":  probeDNS,
+var (
+	Probers = map[string]func(string, http.ResponseWriter, Module) bool{
+		"http": probeHTTP,
+		"tcp":  probeTCP,
+		"icmp": probeICMP,
+		"dns":  probeDNS,
+	}
+	sc = &SafeConfig{
+		C: &Config{},
+	}
+)
+
+func (sc *SafeConfig) reloadConfig(confFile string) (err error) {
+	var c = Config{}
+
+	yamlFile, err := ioutil.ReadFile(confFile)
+	if err != nil {
+		log.Errorf("Error reading config file: %s", err)
+		return err
+	}
+
+	if err := yaml.Unmarshal(yamlFile, &c); err != nil {
+		log.Errorf("Error parsing config file: %s", err)
+		return err
+	}
+
+	sc.Lock()
+	sc.C = &c
+	sc.Unlock()
+
+	log.Infoln("Loaded config file")
+	return nil
 }
 
-func probeHandler(w http.ResponseWriter, r *http.Request, config *Config) {
+func probeHandler(w http.ResponseWriter, r *http.Request, conf Config) {
```

Comment:
This should remain a *Config, so that we're not copying the entire config every time.

The less garbage we create the better.